dev/tests/api-functional/testsuite/Magento/GraphQl/Quote/GetCartTest.php

```diff
@@ -0,0 +1,171 @@
+ * See COPYING.txt for license details.
+ */
+declare(strict_types=1);
+
+namespace Magento\GraphQl\Quote;
+
+use Magento\Integration\Api\CustomerTokenServiceInterface;
+use Magento\Quote\Model\Quote;
+use Magento\Quote\Model\QuoteIdToMaskedQuoteIdInterface;
+use Magento\Quote\Model\ResourceModel\Quote as QuoteResource;
+use Magento\TestFramework\Helper\Bootstrap;
+use Magento\TestFramework\TestCase\GraphQlAbstract;
+
+/**
+ * Test for getting cart information
+ */
+class GetCartTest extends GraphQlAbstract
+{
+    /**
+     * @var CustomerTokenServiceInterface
+     */
+    private $customerTokenService;
+
+    /**
+     * @var QuoteResource
+     */
+    private $quoteResource;
+
+    /**
+     * @var Quote
+     */
+    private $quote;
+
+    /**
+     * @var QuoteIdToMaskedQuoteIdInterface
+     */
+    private $quoteIdToMaskedId;
+
+    /**
+     * @inheritdoc
+     */
+    protected function setUp()
+    {
+        $objectManager = Bootstrap::getObjectManager();
+        $this->quoteResource = $objectManager->create(QuoteResource::class);
+        $this->quote = $objectManager->create(Quote::class);
+        $this->quoteIdToMaskedId = $objectManager->create(QuoteIdToMaskedQuoteIdInterface::class);
+        $this->customerTokenService = $objectManager->get(CustomerTokenServiceInterface::class);
+    }
+
+    /**
+     * @magentoApiDataFixture Magento/Checkout/_files/quote_with_items_saved.php
+     */
+    public function testGetOwnCartForRegisteredCustomer()
+    {
+        $reservedOrderId = 'test_order_item_with_items';
+        $this->quoteResource->load(
+            $this->quote,
+            $reservedOrderId,
+            'reserved_order_id'
+        );
+
+        $maskedQuoteId = $this->quoteIdToMaskedId->execute((int)$this->quote->getId());
+        $query = $this->prepareGetCartQuery($maskedQuoteId);
+
+        $response = $this->sendRequestWithToken($query);
+
+        self::assertArrayHasKey('Cart', $response);
+        self::assertNotEmpty($response['Cart']['items']);
+        self::assertNotEmpty($response['Cart']['addresses']);
+    }
+
+    /**
+     * @magentoApiDataFixture Magento/Checkout/_files/quote_with_items_saved.php
+     */
+    public function testGetCartFromAnotherCustomer()
+    {
+        $reservedOrderId = 'test_order_item_with_items';
+        $this->quoteResource->load(
+            $this->quote,
+            $reservedOrderId,
+            'reserved_order_id'
+        );
+
+
+        $maskedQuoteId = $this->quoteIdToMaskedId->execute((int)$this->quote->getId());
+        $query = $this->prepareGetCartQuery($maskedQuoteId);
+
+        self::expectExceptionMessage("The current user cannot perform operations on cart \"$maskedQuoteId\"");
+
+        $this->graphQlQuery($query);
+    }
+
+    /**
+     * @magentoApiDataFixture Magento/Checkout/_files/active_quote.php
+     */
+    public function testGetCartForGuest()
+    {
+        $reservedOrderId = 'test_order_1';
+        $this->quoteResource->load(
```

Comment:
Why do we need to load Quote?

Reply:
Let me shed some light here.
I use this fixture because we perform tests for this quote. I use this fixture since it assigns maskedQuoteId to the quote. Also, this fixture provides a guest quote (it's not assigned to a customer)
So, the flow is: we load the guest quote, we perform tests for the guest quote.

As for the model loading, we need to get maskedQuoteId somehow but we don't have a quoteId that can be used for quoteIdToMaskedId service (only reservedOrderId). So, I load the model for retrieving quoteId.